src/test/storage/dictionary_segment_test.cpp

```diff
@@ -10,61 +10,106 @@
+  vc_str->append("Steve");
+  vc_str->append("Alexander");
+  vc_str->append("Steve");
+  vc_str->append("Hasso");
+  vc_str->append("Bill");
+
+  std::shared_ptr<BaseSegment> col;
+  resolve_data_type("string", [&](auto type) {
+    using Type = typename decltype(type)::type;
+    col = std::make_shared<DictionarySegment<Type>>(vc_str);
+  });
+  auto dict_col = std::dynamic_pointer_cast<opossum::DictionarySegment<std::string>>(col);
+
+  // Test attribute_vector size
+  EXPECT_EQ(dict_col->size(), 6u);
+
+  // Test dictionary size (uniqueness)
+  EXPECT_EQ(dict_col->unique_values_count(), 4u);
+
+  // Test sorting
+  auto dict = dict_col->dictionary();
+  EXPECT_EQ((*dict)[0], "Alexander");
+  EXPECT_EQ((*dict)[1], "Bill");
+  EXPECT_EQ((*dict)[2], "Hasso");
+  EXPECT_EQ((*dict)[3], "Steve");
+
+  EXPECT_THROW(dict_col->append("Nobody"), std::exception);
+}
+
+TEST_F(StorageDictionarySegmentTest, GetValues) {
+  for (int i = 0; i <= 10; i += 2) vc_int->append(i);
+
+  std::shared_ptr<BaseSegment> col;
+  resolve_data_type("int", [&](auto type) {
+    using Type = typename decltype(type)::type;
+    col = std::make_shared<DictionarySegment<Type>>(vc_int);
+  });
+  auto dict_col = std::dynamic_pointer_cast<opossum::DictionarySegment<int>>(col);
+
+  EXPECT_EQ(dict_col->value_by_value_id((opossum::ValueID)1), 2);
+  EXPECT_THROW(dict_col->value_by_value_id((opossum::ValueID)9), std::exception);
+
+  EXPECT_EQ(dict_col->get(5), 10);
+  EXPECT_THROW(dict_col->get(9), std::exception);
+}
+
+TEST_F(StorageDictionarySegmentTest, LowerUpperBound) {
+  for (int i = 0; i <= 10; i += 2) vc_int->append(i);
+
+  std::shared_ptr<BaseSegment> col;
+  resolve_data_type("int", [&](auto type) {
+    using Type = typename decltype(type)::type;
+    col = std::make_shared<DictionarySegment<Type>>(vc_int);
+  });
+  auto dict_col = std::dynamic_pointer_cast<opossum::DictionarySegment<int>>(col);
+
+  EXPECT_EQ(dict_col->lower_bound(4), (opossum::ValueID)2);
+  EXPECT_EQ(dict_col->upper_bound(4), (opossum::ValueID)3);
+
+  EXPECT_EQ(dict_col->lower_bound(5), (opossum::ValueID)3);
+  EXPECT_EQ(dict_col->upper_bound(5), (opossum::ValueID)3);
+
+  EXPECT_EQ(dict_col->lower_bound(15), opossum::INVALID_VALUE_ID);
+  EXPECT_EQ(dict_col->upper_bound(15), opossum::INVALID_VALUE_ID);
+}
+
+TEST_F(StorageDictionarySegmentTest, SpaceEfficiency) {
+  for (int i = 0; i < 255; i++) vc_int->append(i);
+  vc_int->append(42);
+
+  std::shared_ptr<BaseSegment> col;
+  resolve_data_type("int", [&](auto type) {
+    using Type = typename decltype(type)::type;
+    col = std::make_shared<DictionarySegment<Type>>(vc_int);
+  });
+  auto dict_col = std::dynamic_pointer_cast<opossum::DictionarySegment<int>>(col);
+
+  EXPECT_EQ(dict_col->unique_values_count(), 255u);
+  EXPECT_EQ(dict_col->estimate_memory_usage(), 255 * sizeof(int) + 256 * sizeof(uint8_t));
+  EXPECT_THROW(dict_col->value_by_value_id(static_cast<ValueID>(static_cast<uint8_t>(opossum::INVALID_VALUE_ID))),
```

Comment:
Since the dictionary is a simple vector, you do not have to use a specific size for your index number.

```suggestion
  EXPECT_THROW(dict_col->value_by_value_id(INVALID_VALUE_ID),
```

Additionally, since we are in the opossum namespace, you can get rid of the 'opossum::' prefix.